azurerm/internal/services/storage/storage_account_resource_test.go

```diff
@@ -1954,3 +2037,164 @@ resource "azurerm_storage_account" "test" {
 }
 `, data.RandomInteger, data.Locations.Primary, data.RandomString)
 }
+
+func (r StorageAccountResource) azureFilesIdentityBasedAuthenticationAADDS(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-storage-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "unlikely23exst2acct%s"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+
+  azure_files_identity_based_authentication {
+    directory_service_options = "AADDS"
+  }
+
+  tags = {
+    environment = "production"
+  }
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomString)
+}
+
+func (r StorageAccountResource) azureFilesIdentityBasedAuthenticationAD(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-storage-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "unlikely23exst2acct%s"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+
+  azure_files_identity_based_authentication {
+    directory_service_options = "AD"
+    active_directory {
+      azure_storage_sid    = "S-1-5-21-2400535526-2334094090-2402026252-0012"
+      domain_name          = "adtest.com"
+      domain_sid           = "S-1-5-21-2400535526-2334094090-2402026252-0012"
+      domain_guid          = "aebfc118-9fa9-4732-a21f-d98e41a77ae1"
+      forest_name          = "adtest.com"
+      net_bios_domain_name = "adtest.com"
```

Comment:
Are these dummy values or do they refer to a real forest/domain you have set up for testing?

Reply:
Dummy values.

Reply:
The service has validation for the sid, I could not use zero as replacement.